console/v1alpha1/types_console_plugin.go

```diff
@@ -26,42 +26,80 @@ type ConsolePluginSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
 	// +optional
-	DisplayName string `json:"displayName"`
+	DisplayName string `json:"displayName,omitempty"`
 	// service is a Kubernetes Service that exposes the plugin using a
 	// deployment with an HTTP server. The Service must use HTTPS and
 	// Service serving certificate. The console backend will proxy the
 	// plugins assets from the Service using the service CA bundle.
 	// +kubebuilder:validation:Required
 	// +required
 	Service ConsolePluginService `json:"service"`
-	// proxy is a list of Services that the plugin needs to connect to.
+	// proxy is a list of proxies that describe various service type
+	// to which the plugin needs to connect to.
 	// +kubebuilder:validation:Optional
 	// +optional
-	Proxy ConsolePluginProxy `json:"proxy"`
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
 }
 
 // ConsolePluginProxy holds information on various service types
 // to which console's backend will proxy the plugin's requests.
 type ConsolePluginProxy struct {
-	// services is a list of in-cluster Services that the plugin
-	// will connect to. The Service must use HTTPS. Console backend
-	// exposes the following endpoint in order to proxy communication
-	// between the plugin and the Service:
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type ConsolePluginProxyType `json:"type"`
+	// alias is an proxy name that identifies the plugin's proxy. An alias name
+	// should be unique per proxy. Console backend exposes following endpoint
+	// in order to proxy communication. For example:
 	//
-	// /api/proxy/namespace/<service-namespace>/service/<service-name>:<port-number>/<request-path>?<optional-query-parameters>
+	// /api/proxy/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
 	//
 	// Request example path:
 	//
-	// /api/proxy/namespace/helm/service/helm-charts:8443/releases?limit=10
+	// /api/proxy/acm/search/pods?namespace=openshift-apiserver
 	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +required
+	Alias string `json:"alias"`
+	// service is a in-cluster Services that the plugin will connect to.
+	// The Service must use HTTPS. Console backend exposes the an endpoint
+	// in order to proxy communication between the plugin and the Service.
+	// +kubebuilder:validation:Optional
+	// +optional
```

Comment:
It would be nice to make this required if type is `Service`, but I'm assuming there is no way to conditionally do that.

Reply:
Unfortunatelly it cant :( It needs to be done programatically. Looking at where the additional validation is ATM, cause I know it was located in the *origin* repo. Also if anything operator could do validation as well.

Reply:
It's better to do validation on creation/edit if we can. I'm debating if we shouldn't just make this required for now (since only Service is supported). We can always loosen validation later.